sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubSubPayloadTranslation.java

```diff
@@ -59,6 +60,10 @@ public String getUrn() {
     @Override
     public RunnerApi.FunctionSpec translate(
         AppliedPTransform<?, ?, Unbounded<?>> transform, SdkComponents components) {
+      if (ExperimentalOptions.hasExperiment(
```

Comment:
Would you mind explain a little bit why the current transform override (referred above, override invoked when the experiments not provided) is not sufficient runner v2? From DataflowRunner.java the stacktrack for execution to this point is

DataflowRunner.getOverrides() L552
DataflowRunner.replaceV1Transforms() L1579
DataflowRunner.run() L1193

which means both Dataflow v1 and v2 job submission had the override by default, while non-Dataflow runner always has custom runner (aka Beam provided Pubsub IO)

So there are two questions here
- why the current override working for Dataflow v1 [override can be enabled (default) or disabled (with flag)] but not v2 (persumably always enabled?)?
- is it possible to keep the change inside Dataflow runner, follow the existing pattern there.

Reply:
Since runner v2 uses portable job submission I think that DataflowRunner path is not used, right?

Reply:
Or at least not the replaceV1Transforms path

Reply:
Oh I actually looked at the code and seems that replaceV1Transforms may be intended to run for both versions. In that case I'm not sure yet if it actually runs or if the replacement fails on v2 for another reason.

Reply:
Looks like replaceV1Transforms is only used for v1. The portable job submission proto is done ahead of this. Is it a suitable+better fix to run replaceV1Transforms before creating portablePipelineProto?

Reply:
Thanks for explanation, I see. Both V1 and V2 pipeline protos are constructed in DataflowRunner.run()

- V2 pipeline proto is `portablePipelineProto`, used to write to stagedPipeline
- V1 pipeline proto is `dataflowV1PipelineProto`, used to construct jobSpecification and essentially initiate the job (newJob)

This is unusual and quite easily get confused at first glance... but anywhere here we are. I understand now why the DataflowRunner settings won't affect runner v2 job.

Reply:
I see V1 overwrite is only effective in streaming mode

https://github.com/apache/beam/blob/90cbdd0648e34127fd27809b44e909f23fb37aa1/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java#L551

should we only make the new branch also streaming only or not necessary?

Reply:
I don't think the check is needed since it's replacing the _unbounded_ version, which should already only be used for streaming jobs. But please lmk if there's any concerns with this, thanks!

Reply:
I see, thanks!